app/knight.py

```diff
@@ -0,0 +1,45 @@
+class Knight:
+
+    knights_instances = {}
+
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list = None,
+                 weapon: dict = None,

```

Comment:
Consider initializing `armour`, `weapon`, and `potion` with empty collections (e.g., `[]` or `{}`) instead of `None` to avoid potential errors when calling methods on these attributes.